src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,145 @@
+  private DatabaseReference realtimeDb;
+
+  // Counter that will be updated 
+  Counter counter = new Counter(0);
+
+  @Override
+  public void init() {
+
+    try {
+      // Initialization of reference to Realtime database
+
+      // Building options, include credentials and which DB to use
+      FirebaseOptions options = new FirebaseOptions.Builder()
+          .setCredentials(GoogleCredentials.getApplicationDefault())
+          .setDatabaseUrl("https://quizzy-step-2020.firebaseio.com/")
+          .build();
+
+      // Initialize DB with options
+      FirebaseApp.initializeApp(options);
+
+      // Reference to the whole Realtime DB
+      final FirebaseDatabase database = FirebaseDatabase.getInstance();
+      
+      // Reference to the "counter" object
+      realtimeDb = database.getReference("/users-counter/counter");
+
+      // Add a listener to the counter, executed when counter is changed
+      realtimeDb.addValueEventListener(new ValueEventListener() {
+        @Override
+        public void onDataChange(DataSnapshot dataSnapshot) {
+          counter = dataSnapshot.getValue(Counter.class);
+          System.out.println("Listener executed " + counter.value);
+        }
+      
+        @Override
+        public void onCancelled(DatabaseError databaseError) {
+          System.out.println("The read failed: " + databaseError.getCode());
+        }
+      });
+
+
+      // Initialization of Firestore database
+
+      // Building options, include projectID, access credentials
+      FirestoreOptions firestoreOptions =
+      FirestoreOptions.getDefaultInstance().toBuilder()
+          .setProjectId("quizzy-step-2020")
+          .setCredentials(GoogleCredentials.getApplicationDefault())
+          .build();
+
+      // Get database and store the reference in firestoreDb variable
+      firestoreDb = firestoreOptions.getService();
+
+    } catch(IOException ie) {
+      ie.printStackTrace();
+    }
+
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Writting to the DBs</h1>");
+
+    // Writing to Realtime DB
+
+    // Creating a tempCounter just to show that counter is not updated locally, but in the DB
+    Counter tempCounter = new Counter(counter.value + 1);
+    response.getWriter().println(tempCounter.value);
+
+    // Set in DB with one of the following two options:
+
+    // Update counter asynchronously in Realtime DB with no callback
+    //realtimeDb.setValueAsync(tempCounter);
+
+    // Set value and attach a callback for when it is ready, may not work because of AppEngine thread management?
+    realtimeDb.setValue(tempCounter, new DatabaseReference.CompletionListener() {
+      @Override
+      public void onComplete(DatabaseError databaseError, DatabaseReference databaseReference) {
```

Comment:
probably good practice to also implement onException with your error message too? Or else you'll just miss it! : )

Reply:
I think the CompletitionListener only has method onComplete?https://firebase.google.com/docs/reference/android/com/google/firebase/database/DatabaseReference.CompletionListener#public-method-summary
Or should I look into another method?:D

Reply:
Nice, I think i did a quick search and maybe found https://docs.oracle.com/javaee/7/api/javax/jms/CompletionListener.html